tests/integration/helm/upgrade/util.go

```diff
@@ -227,14 +226,94 @@ func performRevisionUpgradeFunc(previousVersion string) func(framework.TestConte
 			t.Fatal(err)
 		}
 
-		overrideValuesFile = getValuesOverrides(t, revisionValues, s.Hub, s.Tag)
-		helmtest.InstallIstioWithRevision(t, cs, h, tarGzSuffix, revisionChartSuffix, overrideValuesFile)
-		helmtest.VerifyInstallation(t, cs)
+		overrideValuesFile = getValuesOverrides(t, s.Hub, s.Tag, "canary")
+		helmtest.InstallIstioWithRevision(t, cs, h, tarGzSuffix, revisionChartSuffix, overrideValuesFile, false)
+		helmtest.VerifyInstallation(t, cs, false)
 
-		newClient, newServer := sanitycheck.SetupTrafficTest(t, t, revisionLabel)
+		_, newClient, newServer := sanitycheck.SetupTrafficTest(t, t, latestRevisionTag)
 		sanitycheck.RunTrafficTestClientServer(t, newClient, newServer)
 
 		// now check that we are compatible with N-1 proxy with N proxy
 		sanitycheck.RunTrafficTestClientServer(t, oldClient, newServer)
 	}
 }
+
+// performRevisionTagsUpgradeFunc returns the provided function necessary to run inside of a integration test
+// for upgrade capability with stable label revision upgrades
+func performRevisionTagsUpgradeFunc(previousVersion string) func(framework.TestContext) {
+	return func(t framework.TestContext) {
+		cs := t.Clusters().Default().(*kubecluster.Cluster)
+		h := helm.New(cs.Filename(), filepath.Join(previousChartPath, previousVersion))
+
+		//t.ConditionalCleanup(func() {
+		//	err := deleteIstioRevision(h, revisionChartSuffix)
+		//	if err != nil {
+		//		t.Fatalf("could not delete istio: %v", err)
+		//	}
+		//	err = deleteIstio(cs, h)
+		//	if err != nil {
+		//		t.Fatalf("could not delete istio: %v", err)
+		//	}
+		//})
+
+		// install 1.10.0 charts with revision set to "1-10-0"
+		previousRevision := strings.ReplaceAll(previousVersion, ".", "-")
+		overrideValuesFile := getValuesOverrides(t, gcrHub, previousVersion, previousRevision)
+		helmtest.InstallIstioWithRevision(t, cs, h, tarGzSuffix, "-"+previousRevision, overrideValuesFile, true)
```

Comment:
why `"-"+previousRevision` here?

Reply:
I removed this, but the intent is to install with `istiod-REVISION`. It just appends the revision for the helm chart. I modified the code to defer this until it needs to be done.